### PR TITLE
Add a workaround for uninstall on SNO

### PR DESCRIPTION
**Pull request by tiraboschi**
Add a workaround for:
https://bugzilla.redhat.com/show_bug.cgi?id=2031919

TODO: Remove once kubevirt implements a proper fix

Signed-off-by: Simone Tiraboschi <stirabos@redhat.com>

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/upgrade-test-index-image.sh [hunk 1]
```diff
@@ -320,6 +320,13 @@ WH_POD=$( ${CMD} get -n ${HCO_NAMESPACE} pods -l "name=hyperconverged-cluster-we
 ${CMD} logs -n ${HCO_NAMESPACE} "${WH_POD}"
 
 Msg "Brutally delete HCO removing the namespace where it's running"
+# TODO: workaround for https://bugzilla.redhat.com/show_bug.cgi?id=2031919
```
Comment (tiraboschi) [inline hack/upgrade-test-index-image.sh:323]: we should do it only as the last action before removing the namespace

File: hack/upgrade-test-index-image.sh [hunk 2]
```diff
@@ -320,6 +320,13 @@ WH_POD=$( ${CMD} get -n ${HCO_NAMESPACE} pods -l "name=hyperconverged-cluster-we
...
+# remove once fixed
+if [[ "${INFRASTRUCTURETOPOLOGY}" == "SingleReplica" ]]; then
+  echo "Explictly deleting kubevirt apiservice leftovers"
+  ${CMD} delete apiservice v1alpha3.subresources.kubevirt.io || true
```
Comment (erkanerol) [inline hack/upgrade-test-index-image.sh:327]: It is not important but `--ignore-not-found=true` is a useful flag for this kind of stuff. FYI : )

**Comment by nunnatsa** [comment]
/lgtm

waiting with the approval to see if the problem solved.

**Comment by nunnatsa** [comment]
hco-e2e-image-index-aws and hco-e2e-image-index-azure lanes succeeded.
/override ci/prow/hco-e2e-image-index-gcp

hco-e2e-image-index-sno-azure lane succeeded.
/override ci/prow/hco-e2e-image-index-sno-aws

hco-e2e-kv-smoke-azure lane succeeded.
/override ci/prow/hco-e2e-kv-smoke-gcp

**Pull request merged**